src/odb/src/db/dbBlock.cpp

```diff
@@ -1003,11 +1006,17 @@ dbOStream& operator<<(dbOStream& stream, const _dbBlock& block)
   stream << *block._iterm_tbl;
   stream << *block._net_tbl;
   stream << *block._inst_hdr_tbl;
-  stream << *block._inst_tbl;
-  stream << *block._module_tbl;
+  if (db->isSchema(db_schema_db_remove_hash)) {
+    stream << *block._module_tbl;
+    stream << *block._inst_tbl;
+  } else {
+    stream << *block._inst_tbl;
+    stream << *block._module_tbl;
+  }
```

Comment:
Why is the order of tables reversed?  This seems unrelated to the hashes.

Reply:
Tricky to explain... but here goes. To rebuild the hashes (which are contained in the dbModule) containing low level stuff like dbInst from the streamer we have to make sure the dbModule exists so that the has exists before populating the hash. This was the fix that let us rebuild the speed up hashes from the streamer (rather than post populating them after reading whole db).